docs/apps/interchain-accounts/legacy/keeper-api.md

```diff
@@ -0,0 +1,246 @@
+appVersion, err := icatypes.ModuleCdc.MarshalJSON(&icaMetadata)
+if err != nil {
+    return err
+}
+
+feeMetadata := feetypes.Metadata{
+    AppVersion: string(appVersion),
+    FeeVersion: feetypes.Version,
+}
+
+feeEnabledVersion, err := feetypes.ModuleCdc.MarshalJSON(&feeMetadata)
+if err != nil {
+    return err
+}
+
+if err := keeper.icaControllerKeeper.RegisterInterchainAccount(ctx, controllerConnectionID, owner.String(), string(feeEnabledVersion)); err != nil {
+    return err
+}
+```
+
+### `SendTx`
+
+The authentication module can attempt to send a packet by calling `SendTx`:
+
+```go
+// Authenticate owner
+// perform custom logic
+    
+// Construct controller portID based on interchain account owner address
+portID, err := icatypes.NewControllerPortID(owner.String())
+if err != nil {
+    return err
+}
+    
+// Obtain data to be sent to the host chain. 
+// In this example, the owner of the interchain account would like to send a bank MsgSend to the host chain. 
+// The appropriate serialization function should be called. The host chain must be able to deserialize the transaction. 
+// If the host chain is using the ibc-go host module, `SerializeCosmosTx` should be used. 
+msg := &banktypes.MsgSend{FromAddress: fromAddr, ToAddress: toAddr, Amount: amt}
+data, err := icatypes.SerializeCosmosTx(keeper.cdc, []proto.Message{msg})
+if err != nil {
+    return err
+}
+
+// Construct packet data
+packetData := icatypes.InterchainAccountPacketData{
+    Type: icatypes.EXECUTE_TX,
+    Data: data,
+}
+
+// Obtain timeout timestamp
+// An appropriate timeout timestamp must be determined based on the usage of the interchain account.
+// If the packet times out, the channel will be closed requiring a new channel to be created.
+timeoutTimestamp := obtainTimeoutTimestamp()
+
+// Send the interchain accounts packet, returning the packet sequence
+// A nil channel capability can be passed, since the controller submodule (and not the authentication module) 
+// claims the channel capability since ibc-go v6.
+seq, err = keeper.icaControllerKeeper.SendTx(ctx, nil, portID, packetData, timeoutTimestamp)
+```
+
+The data within an `InterchainAccountPacketData` must be serialized using a format supported by the host chain. 
+If the host chain is using the ibc-go host chain submodule, `SerializeCosmosTx` should be used. If the `InterchainAccountPacketData.Data` is serialized using a format not support by the host chain, the packet will not be successfully received.  
+
+### `OnAcknowledgementPacket`
+
+Controller chains will be able to access the acknowledgement written into the host chain state once a relayer relays the acknowledgement. 
+The acknowledgement bytes will be passed to the auth module via the `OnAcknowledgementPacket` callback. 
+Auth modules are expected to know how to decode the acknowledgement. 
+
+If the controller chain is connected to a host chain using the host module on ibc-go, it may interpret the acknowledgement bytes as follows:
```

Comment:
Maybe you can link to the code where we can find the content of the ack, [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/host/keeper/relay.go#L31) if I'm correct?

Reply:
Instead of linking to code, I added a clarification of what the acknowledgement contains. I hope that's ok.